arch/lkl/kernel/setup.c

```diff
@@ -200,7 +200,11 @@ static int __init fs_setup(void)
 {
 	int fd;
 
-	fd = sys_open("/init", O_CREAT, 0700);
+	// Pad '/init' to make sure it's 8 bytes, otherwise KASan would
+	// emit an error. The kernel's strncpy implementation attempts to read
+	// 8 bytes at once and, thus, triggers KASan violation for the 6-byte
+	// string.
+	fd = sys_open("/init\0\0", O_CREAT, 0700);
```

Comment:
Hmm, interesting, I wonder why this does not happen on non LKL arch with KASAN enabled. Created #516 to investigate.

Reply:
Yeah, that's puzzling to me as well. What I think happens is that the KASan wasn't enabled before... I think we forgot to actually enable KASan in the previous commits -- setting kernel config `CONFIG_KASAN_GENERIC=y` enables KASAn (which is addressed in this commit). Without this kernel config the KASan is configured but not actually enabled, I think. At least this was the case for the fuzzer.

Reply:
KASAN was definitely enabled otherwise the kasan tests won't pass.